SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp

```diff
@@ -751,6 +952,94 @@ bool FSpatialGDKEditorToolbarModule::StopSpatialServiceIsVisible() const
+{
+	FModuleManager::LoadModuleChecked<ISettingsModule>("Settings").ShowViewer("Project", "SpatialGDKEditor", "Editor Settings");
+}
+
+void FSpatialGDKEditorToolbarModule::GDKRuntimeSettingsClicked() const
+{
+	FModuleManager::LoadModuleChecked<ISettingsModule>("Settings").ShowViewer("Project", "SpatialGDKEditor", "Runtime Settings");
+}
+
+bool FSpatialGDKEditorToolbarModule::IsNoAutomaticConnectionSelected() const
+{
+	const USpatialGDKEditorSettings* SpatialGDKEditorSettings = GetDefault<USpatialGDKEditorSettings>();
+	return SpatialGDKEditorSettings->SpatialOSNetFlowType == ESpatialOSNetFlow::NoAutomaticConnection;
+}
+
+bool FSpatialGDKEditorToolbarModule::IsLocalDeploymentSelected() const
+{
+	const USpatialGDKEditorSettings* SpatialGDKEditorSettings = GetDefault<USpatialGDKEditorSettings>();
+	return SpatialGDKEditorSettings->SpatialOSNetFlowType == ESpatialOSNetFlow::LocalDeployment;
+}
+
+bool FSpatialGDKEditorToolbarModule::IsCloudDeploymentSelected() const
+{
+	const USpatialGDKEditorSettings* SpatialGDKEditorSettings = GetDefault<USpatialGDKEditorSettings>();
+	return SpatialGDKEditorSettings->SpatialOSNetFlowType == ESpatialOSNetFlow::CloudDeployment;
+}
+
+bool FSpatialGDKEditorToolbarModule::IsSpatialOSNetFlowConfigurable() const
+{
+	return OnIsSpatialNetworkingEnabled() && !(LocalDeploymentManager->IsLocalDeploymentRunning());
+}
+
+void FSpatialGDKEditorToolbarModule::NoAutomaticConnectionClicked()
+{
+	USpatialGDKEditorSettings* SpatialGDKEditorSettings = GetMutableDefault<USpatialGDKEditorSettings>();
+	SpatialGDKEditorSettings->SpatialOSNetFlowType = ESpatialOSNetFlow::NoAutomaticConnection;
+	SpatialGDKEditorSettings->SetUseDevelopmentAuthenticationFlow(false);
+
+	RefreshAutoStartLocalDeployment();
+}
+
+void FSpatialGDKEditorToolbarModule::LocalDeploymentClicked()
+{
+	USpatialGDKEditorSettings* SpatialGDKEditorSettings = GetMutableDefault<USpatialGDKEditorSettings>();
+	SpatialGDKEditorSettings->SpatialOSNetFlowType = ESpatialOSNetFlow::LocalDeployment;
+	SpatialGDKEditorSettings->SetUseDevelopmentAuthenticationFlow(false);
+
+	RefreshAutoStartLocalDeployment();
+}
+
+void FSpatialGDKEditorToolbarModule::CloudDeploymentClicked()
+{
+	USpatialGDKEditorSettings* SpatialGDKEditorSettings = GetMutableDefault<USpatialGDKEditorSettings>();
+	SpatialGDKEditorSettings->SpatialOSNetFlowType = ESpatialOSNetFlow::CloudDeployment;
+
+	TSharedRef<FSpatialGDKDevAuthTokenGenerator> DevAuthTokenGenerator = SpatialGDKEditorInstance->GetDevAuthTokenGeneratorRef();
+	DevAuthTokenGenerator->AsyncGenerateDevAuthToken();
+
+	RefreshAutoStartLocalDeployment();
+}
+
+bool FSpatialGDKEditorToolbarModule::IsLocalDeploymentIPEditable() const
+{
+	const USpatialGDKEditorSettings* SpatialGDKEditorSettings = GetDefault<USpatialGDKEditorSettings>();
+	return GetDefault<UGeneralProjectSettings>()->UsesSpatialNetworking() && (SpatialGDKEditorSettings->SpatialOSNetFlowType == ESpatialOSNetFlow::LocalDeployment);
```

Comment:
Omega Omega Omega Nit: It is probably desirable to do the checks in the following order:

1. SpatialGDKEditorSettings->SpatialOSNetFlowType == ESpatialOSNetFlow::LocalDeployment
2. GetDefault<UGeneralProjectSettings>()->UsesSpatialNetworking()

As one of the checks uses the return value of a function (a black box) we don't know the functions complexity. The other check we know is fast and simple so should probably be done first.

Don't have to do this at all.

Reply:
This is not a hot path and this function call is trivial, so I prefer the logical order of first checking whether Spatial networking is turned on at all (as otherwise all editable options in the dropdown are disabled), and then checking that the correct connection flow for this option is selected.